src/app/space/create/deployments/apps/delete-deployment-modal.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, ElementRef, EventEmitter, Input, OnInit, Output, ViewChild, ViewEncapsulation } from '@angular/core';
+
+import { NotificationType } from 'ngx-base';
+import { ModalDirective } from 'ngx-bootstrap/modal';
+import { Subscription } from 'rxjs';
+import { Environment } from '../models/environment';
+
+import { DeploymentsService } from '../services/deployments.service';
+import { DeploymentCardComponent } from './deployment-card.component';
+
+@Component({
+  host: {
+    class: 'create-dialog'
+  },
+  encapsulation: ViewEncapsulation.None,
+  selector: 'delete-deployment-modal',
+  templateUrl: './delete-deployment-modal.component.html',
+  styleUrls: ['./delete-deployment-modal.component.less']
+})
+export class DeleteDeploymentModal {
+
+  @Input() host: ModalDirective;
+
+  deploymentCardComponent: DeploymentCardComponent;
```

Comment:
Rather than holding a reference to its parent card component, this dialog component should have an `@Output` emitting events to signify when the user has confirmed they do (or don't) want to delete an application. The parent component would then listen for these events and handle them accordingly.